reqwest-middleware/src/middleware.rs

```diff
@@ -31,7 +31,8 @@ use crate::error::{Error, Result};
 ///
 /// [`ClientWithMiddleware`]: crate::ClientWithMiddleware
 /// [`with`]: crate::ClientBuilder::with
-#[async_trait::async_trait]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
 pub trait Middleware: 'static + Send + Sync {
```

Comment:
Do we need to remove the send bound from the middleware too?

Reply:
From memory, I had to remove this bound due to `reqwest::Client` generating non-Send futures in its implementation. `reqwest::Client` itself is still `Send` (at least [the way I'm currently using it](https://github.com/Ekleog/risuto/blob/main/risuto-web/Cargo.toml#L18) it seems to compile and work fine)

That said, we could remove the `Send` bound from the middleware too (wasm is single-thread anyway). It's just that there was no need to do so to make things work, so I tried to add as little variation as possible to the existing code, so that there's fewer special-cases to maintain :)